homeassistant/components/foscam/config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Config flow for foscam integration."""
+from libpyfoscam import FoscamCamera
+from libpyfoscam.foscam import ERROR_FOSCAM_AUTH, ERROR_FOSCAM_UNAVAILABLE
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+)
+from homeassistant.data_entry_flow import AbortFlow
+
+from .const import CONF_STREAM, LOGGER
+from .const import DOMAIN  # pylint:disable=unused-import
+
+STREAMS = ["Main", "Sub"]
+
+DEFAULT_PORT = 88
+
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
```

Comment:
If you have provided a default value it can be Optional
```suggestion
        vol.Optional(CONF_PORT, default=DEFAULT_PORT): int,
```

Reply:
Looked at one of mine and I have the same, e.g. required. Sorry!